test/integration/integration-test.js

```diff
@@ -0,0 +1,193 @@
+/* eslint-disable max-len  */
+/* eslint-disable max-nested-callbacks */
+/* eslint-disable max-params */
+'use strict';
+const _ = require('lodash');
+const assert = require('assert');
+
+const errors = require('../../src/api/common/errors');
+
+const wallet = require('./wallet');
+
+const settingsSpecification = require('../fixtures/settings-specification');
+const trustlineSpecification = require('../fixtures/trustline-specification');
+const payments = require('../fixtures/payments');
+
+const RippleAPI = require('../../src').RippleAPI;
+
+
+const TIMEOUT = 20000;   // how long before each test case times out
+const INTERVAL = 2000;   // how long to wait between checks for validated ledger
+
+
+function verifyTransaction(testcase, hash, type, options, done) {
+  const currentOptions = {
+    minLedgerVersion: options.minLedgerVersion,
+    maxLedgerVersion: testcase.api.getLedgerVersion()
```

Comment:
I think probably the best solution is to make MissingLedgerHistoryError only happen when ledgers are missing prior to the last closed ledger. So in this case I think it should just return NotFoundError.